From: Hilko Bengen <bengen@debian.org>
Date: Sat, 16 Feb 2013 12:36:34 +0100
Subject: out-of-tree build: daemon

- Add include path

    CC     guestfsd-9p.o
  In file included from ../../../daemon/9p.c:32:0:
  ../../../daemon/daemon.h:33:34: fatal error: guestfs-internal-all.h: No such file or directory

- Fix paths for shared source files so that they are found when building out-of-tree.
---
 daemon/Makefile.am |   13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/daemon/Makefile.am b/daemon/Makefile.am
index 024aa6a..6da6082 100644
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -19,8 +19,6 @@ include $(top_srcdir)/subdir-rules.mk
 
 CLEANFILES = stamp-guestfsd.pod
 
-libsrcdir = $(top_builddir)/src
-
 generator_built = \
 	actions.h \
 	stubs.c \
@@ -42,7 +40,7 @@ EXTRA_DIST = \
 	$(BUILT_SOURCES) \
 	guestfsd.pod
 
-$(shared_with_library): %: $(libsrcdir)/%
+$(shared_with_library): %: $(top_srcdir)/src/%
 	rm -f $@
 	ln $< $@
 
@@ -53,10 +51,10 @@ noinst_LIBRARIES = libprotocol.a
 libprotocol_a_SOURCES = guestfs_protocol.c guestfs_protocol.h
 libprotocol_a_CFLAGS = -Wall -Wno-unused -fno-strict-aliasing
 
-$(libsrcdir)/guestfs_protocol.c: force
-	$(MAKE) -C $(libsrcdir) guestfs_protocol.c
-$(libsrcdir)/guestfs_protocol.h: force
-	$(MAKE) -C $(libsrcdir) guestfs_protocol.h
+$(top_builddir)/src/guestfs_protocol.c: force
+	$(MAKE) -C $(top_builddir)/src guestfs_protocol.c
+$(top_builddir)/src/guestfs_protocol.h: force
+	$(MAKE) -C $(top_builddir)/src guestfs_protocol.h
 
 # Build the errnostring perfect hash code.  The generated code has lots
 # of warnings so we must compile it in a separate mini-library.
@@ -196,6 +194,7 @@ guestfsd_LDADD = \
 guestfsd_CPPFLAGS = \
 	-I$(top_srcdir)/gnulib/lib \
 	-I$(top_builddir)/gnulib/lib \
+	-I$(top_srcdir)/src \
 	-I$(top_builddir)/src
 guestfsd_CFLAGS = \
 	$(WARN_CFLAGS) $(WERROR_CFLAGS) \
